packages/core/src/internal/clients/vector/AbstractVectorIndexClient.ts

```diff
@@ -71,39 +93,36 @@ export abstract class AbstractVectorIndexClient
   }
 
   /**
-   * Adds a batch of items into a vector index.
+   * Upserts a batch of items into a vector index.
    *
-   * Adds an item into the index regardless if the ID already exists.
-   * On duplicate ID, a separate entry is created with the same ID.
-   * To deduplicate, first call `deleteItemBatch` to remove all items
-   * with the same ID, then call `addItemBatch` to add the new items.
+   * If an item with the same ID already exists in the index, it will be replaced.
+   * Otherwise, it will be added to the index.
    *
-   * @param {string} indexName - Name of the index to add the items into.
-   * @param {Array<VectorIndexItem>} items - The items to be added into the index.
-   * @returns {Promise<VectorAddItemBatch.Response>} -
-   * {@link VectorAddItemBatch.Success} on success.
-   * {@link VectorAddItemBatch.Error} on error.
+   * @param {string} indexName - Name of the index to upsert the items into.
+   * @param {Array<VectorIndexItem>} items - The items to be upserted into the index.
+   * @returns {Promise<VectorUpsertItemBatch.Response>} -
+   * {@link VectorUpsertItemBatch.Success} on success.
+   * {@link VectorUpsertItemBatch.Error} on error.
    */
-  public async addItemBatch(
+  public async upsertItemBatch(
     indexName: string,
     items: Array<VectorIndexItem>
-  ): Promise<VectorAddItemBatch.Response> {
-    return await this.dataClient.addItemBatch(indexName, items);
+  ): Promise<VectorUpsertItemBatch.Response> {
+    return await this.dataClient.upsertItemBatch(indexName, items);
   }
 
   /**
    * Searches for the most similar vectors to the query vector in the index.
    * Ranks the vectors in the index by maximum inner product to the query vector.
    * If the index and query vectors are unit normalized, this is equivalent to
-   * ranking by cosine similarity. Hence to perform a cosine similarity search,
+   * ranking by cosine similarity. Hence, to perform a cosine similarity search,
```

Comment:
This comment needs to be adjusted to say that it uses whatever similarity metric the user selected.